pkg/pod/status.go

```diff
@@ -140,9 +141,98 @@ func MakeTaskRunStatus(logger *zap.SugaredLogger, tr v1beta1.TaskRun, pod *corev
 
 	trs.TaskRunResults = removeDuplicateResults(trs.TaskRunResults)
 
+	if resolvedTaskSpec != nil {
```

Comment:
the "pod" package doesn't seem like the best place for this functionality; it more likely belongs in the reconciler code.

Reply:
oh okk, I did put them in the reconcile before but the tests seem very hard to add. 😢 . Will try again!

Reply:
Curious what is making adding tests challenging? I think there may be some opportunity here to create a package specifically for results or substitution or something like that

Reply:
Did you mean 1) put this code inside the `reconcile` function? Or 2) wrap them as a function and call this function inside the `reconcile`? For 1) the reconcile test code is pretty large, for 2) I think we can test this function separately without testing it from `reconcile`?

Reply:
let's do 2, but the pod package isn't the right place for this IMO

Reply:
oh ok, got it!

Reply:
moved them to validate_resources.go